src/libmongoc/tests/mock_server/request.c

```diff
@@ -556,8 +559,20 @@ request_matches_msg (const request_t *request,
 
       /* make sure the pattern is reasonable, e.g. that we didn't pass a string
        * instead of a bson_t* by mistake */
-      BSON_ASSERT (bson_validate (
-         pattern, BSON_VALIDATE_EMPTY_KEYS | BSON_VALIDATE_UTF8, NULL));
+      if (!bson_validate_with_error (pattern,
+                                     BSON_VALIDATE_EMPTY_KEYS |
+                                        BSON_VALIDATE_UTF8,
+                                     &bson_error)) {
+         fprintf (stderr,
```

Comment:
Capturing the error from `bson_validate` is a good idea. I suggest using `test_error`, which prints the filename and line number before aborting
```suggestion
         test_error (
```

Reply:
That's a good idea! Done.